aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java

```diff
@@ -222,9 +235,19 @@ private List<Artifact> getContentPackages() throws MojoExecutionException {
             if (result.isEmpty()) {
                 throw new MojoExecutionException("No content packages found for project.");
             }
-            getLog().info("Found content packages from dependencies: " + result);
-            return result;
+            getLog().info("Using content packages from dependencies: " + result);
+        }
+        if (additionalContentPackageArtifacts != null) {
+            additionalContentPackageArtifacts.stream()
+                .map(ArtifactId::fromMvnId)
+                .map(a -> new ArtifactId(a.getGroupId(), a.getArtifactId(), a.getVersion(), a.getClassifier(), Constants.PACKAGING_ZIP))
```

Comment:
Why do we need to hard-code the type? Couldn't this be specified as part of the maven coordinates?

Reply:
Maybe I'm misreading the patch, but can't one just specify
 com.foo:foopackage:zip:1.2.0
 as an additional content package artifact?
 Instead of just "com.foo:foopackage:1.2.0"

Reply:
Maven type allow either extension values directly or type indirection values. The latter is "content-package" the former is "zip". Not allowing type indirection in Maven coordinates feels incorrect.

Reply:
but your patch does not allow that either - you just assume that it is packaging zip - regardless of what is specified in the parameter

Reply:
That is kind of allowing it :-) because I don't care what the user enters.

Reply:
I think we can use artifactHandlerManager here to get the extension instead - I wouldn't change getOrResolveArtifact for now as that is used in several places

Reply:
It is wrong in all those places ;-) I just checked `aem-analyse` with dependencies of type `content-package`. That does only work when it finds a result in https://github.com/adobe/aemanalyser-maven-plugin/blob/0c905bfe4ab2bbc7c05cb91f7a60e9123c186672/aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AbstractAemMojo.java#L108. But looking up the artifact with extension `content-package` will always fail: https://github.com/adobe/aemanalyser-maven-plugin/blob/0c905bfe4ab2bbc7c05cb91f7a60e9123c186672/aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AbstractAemMojo.java#L110

Reply:
As mentioned I wouldnt do this at the central place as this might have an effect on other parts, and indeed with your PR now the ITs fail:
[INFO] Building: content-package-files-test/pom.xml
[INFO]   The build exited with code 1. See /home/cziegeler/workspaces/aem/aemanalyser-maven-plugin/aemanalyser-maven-plugin/target/it/content-package-files-test/build.log for details.
[INFO]           content-package-files-test/pom.xml ............... FAILED (2.506 s)
[INFO] Building: sling-initial-content-packages-test/pom.xml
[INFO]   The build exited with code 1. See /home/cziegeler/workspaces/aem/aemanalyser-maven-plugin/aemanalyser-maven-plugin/target/it/sling-initial-content-packages-test/build.log for details.
[INFO]           sling-initial-content-packages-test/pom.xml ...... FAILED (2.230 s)
[INFO] Building: wknd-test/pom.xml
[INFO]   The build exited with code 1. See /home/cziegeler/workspaces/aem/aemanalyser-maven-plugin/aemanalyser-maven-plugin/target/it/wknd-test/build.log for details.
[INFO]           wknd-test/pom.xml ................................ FAILED (2.254 s)

Reply:
on it...

Reply:
Now null safe with https://github.com/adobe/aemanalyser-maven-plugin/pull/297/commits/6c34f05e1650bef74148e3c1a5e02c4e7edc6b9f.

Reply:
Thanks, I still get a NPE in the IT tests :)

Reply:
Even null safer now with [bdc27bb](https://github.com/adobe/aemanalyser-maven-plugin/pull/297/commits/bdc27bbb29c01c2dddc8373e9d80515688e4bcff) ;-)